src/Microsoft.PowerShell.Commands.Utility/commands/utility/group-object.cs

```diff
@@ -286,12 +318,71 @@ public SwitchParameter NoElement
                 }
             }
         }
-        private void WriteNonTerminatingError(Exception exception, string resourceIdAndErrorId,
-      ErrorCategory category)
+
+        /// <summary>
+        /// Utility function called by Group-Object to create Groups.
+        /// </summary>
+        /// <param name="currentObjectEntry">Input object that needs to be grouped.</param>
+        /// <param name="noElement">True if we are not accumulating objects.</param>
+        /// <param name="groups">List containing Groups.</param>
+        /// <param name="groupInfoDictionary">Dictionary used to keep track of the groups with hash of the property values being the key.</param>
+        /// <param name="orderByPropertyComparer">The Comparer to be used while comparing to check if new group has to be created.</param>
+        internal static void DoOrderedGrouping(
+            OrderByPropertyEntry currentObjectEntry,
+            bool noElement,
+            List<GroupInfo> groups,
+            Dictionary<object, GroupInfo> groupInfoDictionary,
+            OrderByPropertyComparer orderByPropertyComparer)
+        {
+            var currentObjectOrderValues = currentObjectEntry.orderValues;
+            if (currentObjectOrderValues != null && currentObjectOrderValues.Count > 0)
+            {
+                object currentTupleObject = PSTuple.ArrayToTuple(currentObjectOrderValues);
+
+                if (groupInfoDictionary.TryGetValue(currentTupleObject, out var currentGroupInfo))
+                {
+                    // add this inputObject to an existing group
+                    currentGroupInfo?.Add(currentObjectEntry.inputObject);
```

Comment:
same as above, when is currentTobpleObject present but currentGroupInfo is null.

Reply:
Yeah, I'm removing the null checks.